src/new/wallet.js

```diff
@@ -3129,6 +3130,63 @@ class HathorWallet extends EventEmitter {
     });
     return this.handleSendPreparedTransaction(transaction);
   }
+
+  /**
+   * @typedef {Object} CreateOnChainBlueprintTxOptions
+   * @property {string?} [pinCode] PIN to decrypt the private key.
+   */
+
+  /**
+   * Create and send an on chain blueprint transaction
+   *
+   * @param {string} code Blueprint code in utf-8
+   * @param {string} address Address that will be used to sign the on chain blueprint transaction
+   * @param {CreateOnChainBlueprintTxOptions} [options]
+   *
+   * @returns {Promise<OnChainBlueprint>}
+   */
+  async createAndSendOnChainBlueprintTransaction(code, address, options = {}) {
+    const sendTransaction = await this.createOnChainBlueprintTransaction(code, address, options);
+    return sendTransaction.runFromMining();
+  }
+
+  /**
+   * Create an on chain blueprint transaction and return the SendTransaction object
+   *
+   * @param {string} code Blueprint code in utf-8
+   * @param {string} address Address that will be used to sign the on chain blueprint transaction
+   * @param {CreateOnChainBlueprintTxOptions} [options]
+   *
+   * @returns {Promise<SendTransaction>}
+   */
+  async createOnChainBlueprintTransaction(code, address, options) {
+    if (await this.storage.isReadonly()) {
+      throw new WalletFromXPubGuard('createOnChainBlueprintTransaction');
```

Comment:
There are some annotations for some lines in this file, should we add tests for those cases?

Reply:
Added the tests https://github.com/HathorNetwork/hathor-wallet-lib/pull/811/commits/e97de5f31fcf94f95762125b32f724e019350732